tasks/pkg-windows.yml

```diff
@@ -76,11 +118,16 @@
   when: datadog_windows_ddagentuser_password | default('', true) | length > 0
   no_log: true
 
-- name: Uninstall agent to update optional features
+- name: Uninstall installed Datadog Agent to update optional features or downgrade or because requested
   ansible.windows.win_package:
-    path: "{{ agent_download_msi_result.dest }}"
+    product_id: "{{ item }}"
     state: absent
-  when: not agent_datadog_skip_install and agent_datadog_force_reinstall
+  loop: "{{ agent_datadog_windows_installed_product_ids }}"
+  when:
+    - not agent_datadog_skip_install
+    - agent_datadog_force_reinstall
+    - agent_datadog_windows_installed_product_ids is defined
```

Comment:
I'm not sure if this is a case we need to take into account, but what happens if we try to force reinstall on a clean host without the Agent present (ie `agent_datadog_windows_installed_product_ids` isn't defined)? Should we fail gracefully or install the Agent anyway?